pkg/controller/certificates/groupapprove.go

```diff
@@ -72,7 +72,7 @@ func (cc *groupApprover) AutoApprove(csr *certificates.CertificateSigningRequest
 	if len(x509cr.DNSNames)+len(x509cr.EmailAddresses)+len(x509cr.IPAddresses) != 0 {
 		return csr, nil
 	}
-	if !hasExactUsages(csr, kubeletClientUsages) {
+	if !hasExactUsages(csr, kubeletClientUsages) && !hasExactUsages(csr, kubeletServerUsages) {
```

Comment:
During bootstrapping, the client cert contains the common name `system:node:( nodename )`. When it requests a serving cert from the certificates API, it'll be authenticated as the username `system:node:( nodename )`. When approving these requests, I'd expect that a node would only be able to request serving certs for it's nodename, and not others.

Actually, after the groupapprover is used to grant the client cert, the kubelet can use that cert and wont be part of the bootstrapping group after. This was brought up by @deads2k in the original client cert bootstrapping PR: https://github.com/kubernetes/kubernetes/pull/30094#issuecomment-238690754

cc @kubernetes/sig-auth-pr-reviews